src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -1344,22 +1345,22 @@ private string[] SplitInternal(string separator, string[] separators, int count,
 
             if (!singleSeparator && (separators == null || separators.Length == 0))
             {
-                return SplitInternal((char[])null, count, options);
+                return SplitInternal(default(ReadOnlySpan<char>), count, options);
             }
 
             if ((count == 0) || (omitEmptyEntries && Length == 0))
             {
                 return Array.Empty<string>();
             }
 
-            if (count == 1 || (singleSeparator && separator.Length == 0))
+            if (count == 1 || (singleSeparator && separator!.Length == 0))
```

Comment:
@jaredpar, is there benefit in us tracking places where the compiler's flow analysis was unable to infer non-nullness, i.e. would it help the compiler team to refine the analysis?  Or it's not worthwhile / it could always be investigated later via an analyzer that flagged all cases where the null-forgiveness operator was used?

Reply:
I think it would be helpful to see cases that come up repeatedly for sure.